File: leonardo_rifeli/app/Coffee.php
<?php

namespace App;

class Coffee {

    private $isFull = false;
    
    public function isEmpty()
    {
        return !$this->isFull;
    }
    
    public function refill()
    {
        if(!$this->isFull) {
            $this->isFull = true;
        }

        return true;
    }
    
    public function drink()
    {
        if(!$this->isFull) {
            return false;
        }

        $this->isFull = false;

        return true;
    }
    
}

File: marcelo_camargo/Coffee.hs
module Coffee (refill, drink, mug, programmersLife, Coffee, hasCoffee) where

data Coffee = Coffee {
  hasCoffee :: Bool
} deriving (Show, Eq)

refill :: Coffee -> Coffee
refill _ = Coffee { hasCoffee = True }

drink :: Coffee -> Coffee
drink _ = Coffee { hasCoffee = False }

mug :: Bool -> Coffee
mug b = Coffee { hasCoffee = b }

programmersLife :: Coffee -> IO Coffee
programmersLife coffee = let fullCoffee = refill coffee
                             emptyCoffee = drink fullCoffee
                         in return emptyCoffee
